lib/ansible/galaxy/collection/__init__.py

```diff
@@ -586,6 +594,180 @@ def build_collection(collection_path, output_path, force):
     return collection_output
 
 
+def remove_collections(collections, search_paths, no_deps, force):
+    """Deletes the directories of installed collections that match the requirements list
+
+    :param collections: A list of collection requirements to find and remove
+    :param search_paths: A list of paths to search for the collections
+    :param no_deps: A boolean to indicate whether dangling dependences should be ignored instead of removed
+    :param force: A boolean to bypass prompting when multiple matching collections are found
+    """
+    # Find all collections in all paths to ensure the deps of collections we remove are only removed if they will become unused
+    everything = []
+    for search_path in search_paths:
+        everything.extend(find_existing_collections(search_path, fallback_metadata=True))
+
+    # Find collections and unused dependencies
+    collections_to_remove = get_matching_collections(collections, everything, force)
+    if no_deps:
+        deps_still_needed = {}
+        deps_to_remove = []
+    else:
+        deps_of_removed_collections = get_dependencies(collections_to_remove, everything)
+        deps_still_needed = get_parents(deps_of_removed_collections, collections_to_remove, everything)
+        deps_to_remove = set(deps_of_removed_collections) - set(deps_still_needed.keys())
+
+    # Remove stuff
+    if collections_to_remove:
+        display.display("Removing requested collections:")
+        for collection in collections_to_remove:
+            collection.remove()
+    else:
+        display.display("No collections to remove")
+
+    if deps_to_remove:
+        display.display("Removing unused dependencies:")
+        for dep in deps_to_remove:
+            dep.remove()
+    for dep in deps_still_needed:
+        needed_by = [to_text(needs_dep, errors='surrogate_or_strict') for needs_dep in deps_still_needed[dep]]
+        display.vvvv("Skipping removing dependency '%s' because it is still used by the installed collections: %s" % (
+            to_text(dep, errors='surrogate_or_strict'), ', '.join(needed_by))
+        )
+
+
+def get_matching_collections(requirements, installed_collections, force):
+    """Gets the list of collections that match the requirements list
```

Comment:
Also for consistency
```suggestion
    """Get the list of collections that match the requirements list.
```